src/query_bed.py

```diff
@@ -6,13 +6,55 @@
 from bed import (
     read_bed_file, print_line, BedLine
 )
+from bounds import lower_bound
 
 
 def extract_region(features: list[BedLine],
                    start: int, end: int) -> list[BedLine]:
     """Extract region chrom[start:end] and write it to outfile."""
-    return []  # FIXME: We want the actual region, not an empty list!
+    # FIXME: We want the actual region, not an empty list!
 
+    list_region = []
+    for elements in features:
+        list_region.append(elements[1])
```

Comment:
This is not so great, though...

The reason we use binary search is such that we can find the interval we want in $O(\log n)$. But you start by copying the start positions in the entire region, which takes $O(n)$ time. You are spending time $O(n)$, and you would have been better off just using linear search, then.

The right thing to do is to search in the features, and all you had to do was get the start positions out of the values _when you looked at a line_, not extracting all the start positions for each query.